libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -174,13 +175,11 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
                         request(callback, modifiedRerouteOption)
                     }
                     is RouteOptionsUpdater.RouteOptionsResult.Error -> {
-                        mainJobController.scope.launch {
```

Comment:
Why is this an issue in the first place?

Reply:
By the way, I noticed it in my test, it's not related to this PR but I wanted to discuss it anyway.
When we invoke `interrupt`, we can still get `onRoutesReady` for this request **after** the `interrupt` invocation. I'm not sure if it's right. It breaks the test but I can look for options to fix the test but is it the correct way to approach it? I mean we interrupt the reroute request when we receive new routes. And we can old reroute after setting new routes, I don't think it's right. I think it's related to the fact that `directionsSession.cancelRouteRequest` does not synchronously invoke `onCancel` but I think we get handle this on `MapboxRerouteController` side. WDYT?

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/6693